lib/xdrgen/generators/rust/src/types.rs

```diff
@@ -2767,3 +2816,335 @@ mod test {
         assert_eq!(v.to_option(), Some(1));
     }
 }
+
+#[cfg(all(test, any(feature = "std", feature = "embedded_io")))]
```

Comment:
I added some tests to verify whether the behavior of embedde io meets expectations.